exercises/dnd-character/.meta/src/reference/java/DnDCharacter.java

```diff
@@ -22,12 +23,14 @@
     }
 
     int ability() {
-        int[] scores = rollDices();
-        int scoreSum = 0;
-        for (int i = 1; i < 4; i++) {
-            scoreSum += scores[i];
-        }
-        return scoreSum;
+        // Throw four dice
+        int a = random.nextInt(6) + 1;
+        int b = random.nextInt(6) + 1;
+        int c = random.nextInt(6) + 1;
+        int d = random.nextInt(6) + 1;
+        // Sum the largest three
+        int min = Math.min(Math.min(a, b), Math.min(c, d));
+        return a + b + c + d - min;
```

Comment:
The was nothing inherently wrong with the previous implementation (other than the `Random` object being created every time `ability` is called, as discussed previously here https://github.com/exercism/java/pull/1623#discussion_r410287158). This simply was the one I was using during all of my tests.

I did start out with the reference as well as a few other possible implementations, but as I started testing with more and more iterations, runtime does start to matter. This solution has about a 40% speedup compared to the sorting-array method, and about a 400% speedup compared to a stream based approach.

Now I know that performance isn't probably the most important thing for our reference solutions, but I would argue that this solution is actually more readable and easier to grasp quickly than the previous one. For example it is not clear from the name of the method `rollDices` that it returns _four_ dice in _ascending_ order. Further it isn't necessary to sort the entire array we only need the smallest element. Thus the intention is clearer.

So I would argue in favor of using this implementation in the reference solution.

Reply:
Works for me.